### ISSUE TITLE
addition of a extra vector

### ISSUE THREAD

**Issue by GlastonburyC** [WeidiXie/cell_counting_v2#6]
Hi I would like to modify any one of the counting networks to have an additional vector input and therefore an additional output (MSE loss would be fine).

I think the network may learn several cell characteristics so I would like to try learning the diameter of the cells as well. Whats the implementation I should try? I was thinking adding a merge layer after u_net and then a dense layer(1).

**Comment by WeidiXie** [comment]
Well, I think I need to see what kind of data you are trying to do, 
because I assume not all the cells are of the same shape and same size, right ?

Or you want  to predict a single size (mean size) for all cells ?

Weidi

**Comment by GlastonburyC** [comment]
Thanks for the speedy reply.

You're correct, the cells are close together and of different shapes and sizes. Initially I was going to estimate a mean cell diameter. But I would like to know if you think it's better / possible to estimate perhaps a distribution of cell sizes per image.

But a good starting point would be mean diameter and mean area, both of which I can annotate.

**Comment by WeidiXie** [comment]
Hi, what kind of annotation do you have?
do you have the dot annotation for each cell ?
if you do, why not try to use level set, 
the data looks in a very good condition in level-set assumption, boundaries are clear and sharp, inside region are smooth.

**Comment by GlastonburyC** [comment]
I have the dot annotations done, and I will also have a vector of diameters and areas for each of the images.

Could you point me towards some information for using level-set in this context?

**Comment by WeidiXie** [comment]
Have you tried watershed for segmentation ?
Use your dot annotation as seed.

**Comment by GlastonburyC** [comment]
I have tried Watershed before having the annotations. But using the dots as seeds is a good idea. 

The thing is whilst the image above is clearly easy to segment, some slides will have many small cells that are less well defined. I would have thought these FCN approaches would have a lower error rate?

**Comment by WeidiXie** [comment]
Yes, 
I believe learning-based method will produce good performance, 
however, I think for this data, watershed should give you a pretty strong baseline already.

**Comment by GlastonburyC** [comment]
Thanks. So do you think it's ok to merge the U-net with a dense layer to have an additional input and output?

Much like [this](https://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models) in Keras.

**Comment by WeidiXie** [comment]
Well, if you simply want to (roughly) predict the cell size, 
I would not do that.

I would build a relationship between this size information with the dot annotation, for instance, I may first try to build a bridge between the gaussian sigma and cell size. 

That's just my very initial idea.  : - >

**Comment by WeidiXie** [comment]
During prediction, you should be able to infer the cell size from the predicted gaussian shape density map.
